### PR TITLE
WIP: Add scheduler support to ApplicationSLA model

### PR DESCRIPTION
**Pull request by joshi-keyur**
### Description of the Change

Added schedulerName to ApplicationSLA and defaulted it to "fenzo".

### CODE REVIEW

File: titus-api/src/main/java/com/netflix/titus/api/endpoint/v2/rest/representation/ApplicationSlaRepresentation.java [hunk 1]
```diff
@@ -106,6 +112,8 @@ public ApplicationSlaRepresentation(@JsonProperty("appName") String appName,
         this.instanceNetworkMbs = instanceNetworkMbs;
         this.instanceCount = instanceCount;
         this.reservationUsage = reservationUsage;
+        this.schedulerName = schedulerName;
+
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/endpoint/v2/rest/representation/ApplicationSlaRepresentation.java:116]: Empty line

File: titus-api/src/main/java/com/netflix/titus/api/store/v2/ApplicationSlaStoreCache.java [hunk 2]
```diff
@@ -50,12 +55,17 @@ public ApplicationSlaStoreCache(ApplicationSlaStore delegate) {
...
+        this.cacheBySchedulerName = loadSchedulerMapCache(this.cache);
         return Observable.empty();
     }
 
     @Override
     public Observable<Void> create(ApplicationSLA applicationSLA) {
-        return delegate.create(applicationSLA).doOnCompleted(() -> cache.put(applicationSLA.getAppName(), applicationSLA));
+        return delegate.create(applicationSLA).doOnCompleted(() ->
+        {
+            cache.put(applicationSLA.getAppName(), applicationSLA);
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/store/v2/ApplicationSlaStoreCache.java:66]: Non atomic operation. We should synchronize the update here.

File: titus-api/src/main/java/com/netflix/titus/api/store/v2/ApplicationSlaStoreCache.java [hunk 3]
```diff
@@ -82,12 +100,23 @@ public ApplicationSlaStoreCache(ApplicationSlaStore delegate) {
 
     @Override
     public Observable<Void> remove(String applicationName) {
-        return delegate.remove(applicationName).doOnCompleted(() -> cache.remove(applicationName));
+        return delegate.remove(applicationName).doOnCompleted(() -> {
+            ApplicationSLA result = cache.remove(applicationName);
```
Comment (tbak) [inline titus-api/src/main/java/com/netflix/titus/api/store/v2/ApplicationSlaStoreCache.java:104]: Some as above, we should synchronize here

File: titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraApplicationSlaStore.java [hunk 4]
```diff
@@ -91,6 +92,11 @@ public CassandraApplicationSlaStore(CassandraStoreConfiguration configuration,
         );
     }
 
+    @Override
+    public Observable<ApplicationSLA> findBySchedulerName(String schedulerName) {
+        return findAll().filter(applicationSLA -> Strings.isNullOrEmpty(schedulerName) || applicationSLA.getSchedulerName().equals(schedulerName));
```
Comment (tbak) [inline titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraApplicationSlaStore.java:97]: Shouldn't we check here for fenzo?

Comment (joshi-keyur) [reply titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraApplicationSlaStore.java:97]: Is your point about when the input scheduerName isNullOrEmpty?

File: titus-api/src/main/java/com/netflix/titus/api/store/v2/ApplicationSlaStoreCache.java (hunk 2 above)
Comment (joshi-keyur) [reply titus-api/src/main/java/com/netflix/titus/api/store/v2/ApplicationSlaStoreCache.java:66]: Good catch. Thanks

File: titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraApplicationSlaStore.java (hunk 4 above)
Comment (tbak) [reply titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraApplicationSlaStore.java:97]: Yes. Empty == 'fenzo'

Comment (joshi-keyur) [reply titus-ext/cassandra/src/main/java/com/netflix/titus/ext/cassandra/store/CassandraApplicationSlaStore.java:97]: Added javadoc explaining we return all ApplicationSLA objects when scheduler is null to support backward compatibility.

**Pull request merged**